### PR TITLE
TODO cleanup PT 1

### PR DESCRIPTION
**Pull request by ith-harvey**
Removed three TODOs:
* One was irrelevant, didn't need to be there confirmed with @MikeHathaway 
* Checking the parent index sum after removal
* created utility approval wrapper for borrowers when running repay

### CODE REVIEW

File: tests/forge/interactions/ERC721TakeWithExternalLiquidity.sol [hunk 1]
```diff
@@ -9,99 +9,85 @@ import { NFTCollateralToken } from '../utils/Tokens.sol';
...
         tokenIdsToAdd[1] = 3;
-        changePrank(_borrower);
-        _ajnaPool.drawDebt(_borrower, 1_999 * 1e18, 3232, tokenIdsToAdd);
 
-        // borrower2 adds collateral and borrows a trivial amount
-        tokenIdsToAdd[0] = 4;
-        tokenIdsToAdd[1] = 5;
-        changePrank(_borrower2);
-        _ajnaPool.drawDebt(_borrower2, 5 * 1e18, 3232, tokenIdsToAdd);
+        _drawDebt({
```
Comment (ith-harvey) [inline tests/forge/interactions/ERC721TakeWithExternalLiquidity.sol:51]: Removed the second borrower -> `borrower2` from the test as the previous `TODO` had suggested.

**Review by EdNoepel — COMMENTED** [review]
Please update submodules.

File: tests/forge/ERC721Pool/ERC721PoolCollateral.t.sol [hunk 2]
```diff
@@ -252,7 +252,6 @@ contract ERC721PoolCollateralTest is ERC721HelperContract {
         );
     }
 
-    // TODO: finish implementing
```
Comment (ith-harvey) [inline tests/forge/ERC721Pool/ERC721PoolCollateral.t.sol:255]: Spoke with Mike, not relevant anymore

File: tests/forge/ERC721Pool/ERC721DSTestPlus.sol [hunk 3]
```diff
@@ -642,6 +642,31 @@ abstract contract ERC721HelperContract is ERC721DSTestPlus {
...
+        uint256 amountRepaid,
+        uint256 collateralToPull,
+        uint256 newLup
+    ) internal {
+        changePrank(from);
+        _quote.approve(address(_pool), amountToRepay);
+        _repayDebt(from, borrower, amountToRepay, amountRepaid, collateralToPull, newLup);
+    }
+
+    function _approveAndRepayDebtNoLupCheck(
```
Comment (MikeHathaway) [inline tests/forge/ERC721Pool/ERC721DSTestPlus.sol:659]: Not a big deal, but can also use the new `_calculateLup` helper method to take the expected debt before - amountRepaid and get the resulting lup so we don't have to maintain separate `...NoLupCheck` methods.

File: tests/forge/AjnaRewards.t.sol [hunk 4]
```diff
@@ -699,12 +652,14 @@ contract AjnaRewardsTest is DSTestPlus {
             pool:              address(_poolOne),
             tokenId:           tokenIdOne,
             claimedArray:      _epochsClaimedArray(1, 0),
-            reward:            0.227347187766462422 * 1e18,
```
Comment (MikeHathaway) [inline tests/forge/AjnaRewards.t.sol]: Why was this test changed? It seems to no longer be hitting the cap, and thus resulting in less code coverage

**Review by MikeHathaway — CHANGES_REQUESTED** [review]
Changes to `testClaimRewardsCap` should be reverted since they broke the purpose of the test, reducing code coverage from no longer hitting the cap.

Also need to run `git submodule sync && git submodule update` as clones-with-immutable-args was replaced with a different upstream.

File: tests/forge/AjnaRewards.t.sol (hunk 4 above)
Comment (ith-harvey) [reply tests/forge/AjnaRewards.t.sol]: I misunderstood what the test was doing, apologies : )

Comment (MikeHathaway) [reply tests/forge/AjnaRewards.t.sol]: No worries! thanks for the revert

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**